### PR TITLE
Update to non-deprecated protobufs

### PR DESCRIPTION
**Pull request by mkeeler**
I really just needed to update to use google.golang.org/protobuf instead of golang/protobuf. While in here I also did the following:

1. Updated proto build system to use buf (generation, linting & formatting)
2. Added github actions to run go tests, go linter and buf linter
3. Fixed linting warnings/errors
4. Upgraded dependencies.

I don't think this should be considered a breaking change as I imported it into Consul and all our chunking tests have run without issues or other code modifications. The main advantage to all of this is that the generated code now implements the necesssary interfaces for protoreflect to work and thus most of the non-deprecated go protobuf tooling.

### CODE REVIEW

File: chunking.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func (i *InmemChunkStorage) GetChunks() (ChunkMap, error) {
 
 func (i *InmemChunkStorage) RestoreChunks(chunks ChunkMap) error {
 	// If passed in explicit emptiness, set state to empty
-	if chunks == nil || len(chunks) == 0 {
```
Comment (mkeeler) [inline chunking.go:92]: Linter showed that this was unnecessary

File: types/types.proto [hunk 2]
```diff
@@ -1,23 +1,21 @@
 syntax = "proto3";
 
-option go_package = "types";
```
Comment (mkeeler) [inline types/types.proto:3]: Not needed when using buf managed mode.

File: raft_test.go [hunk 3]
```diff
@@ -58,6 +58,7 @@ func TestRaftStability_Large_Values(t *testing.T) {
 	var wg sync.WaitGroup
 	var numStarted uint32
 	for _, data := range dataBlocks {
+		data := data
 		// Should be able to apply
 		wg.Add(1)
 		go func() {
```
Comment (dhiaayachi) [inline raft_test.go]: feel free to ignore as it's in a testing file but a more conventional way to fix this warning would be to pass `data` as a param to the anonymous func.

**Review by dhiaayachi — APPROVED** [review]
LGTM!
I have a small nitpick, but nothing blocking.

File: raft_test.go (hunk 3 above)
Comment (mkeeler) [reply raft_test.go]: I can do that. This was just me blindly following the linter recommendation but passing it as an arg to the function seems better.

**Pull request merged**